pkg/config/system.go

```diff
@@ -378,8 +378,6 @@ func DecodeKeyIntoZoneIDAndSuffix(
 		} else {
 			objectID = keys.SystemRangesID
 		}
-	} else if bytes.HasPrefix(key, keys.TenantPrefix) {
-		objectID = keys.TenantsRangesID
```

Comment:
I'm a bit unsure about this change.  I think this condition might be outdated.  The problem I ran into was that if the codec belongs to a secondary tenant, the function always uses `keys.TenantsRangesID` for the `objectID`.  This messed up the zone config returned by `getZoneConfig` (and took me way too long to figure out!).

I could also fix this by keeping the condition but only using it for `code.ForSystemTenant`.  But I'm wondering if these pseudo tenant zone configs are still relevant.  Is there a risk in removing this for backward compatibility?

Any help confirming this would be appreciated!